File: backend/config-services/src/main/java/com/microservices/configservices/service/PostService.java
package com.microservices.configservices.service;

import java.util.List;
import java.util.UUID;
import com.microservices.configservices.payload.request.PostRequest;
import com.microservices.configservices.payload.response.PostResponse;

public interface PostService {

    PostResponse create(PostRequest postRequest);

    void setImage(UUID id, String image);

    void switchStatus(UUID id);

    void trash(UUID id);

    PostResponse getById(UUID id);

    List<PostResponse> getAll();

    PostResponse update(UUID id, PostRequest postRequest);

    PostResponse delete(UUID id);
}

File: frontend/reactjs-app/src/services/PostService.js
import httpAxios from '../httpAxios';

const Postservice = {
    create: (postData) => {
        return httpAxios.post(`config-services/api/posts/create`, postData);
    },
    setImage: (data) => {
        return httpAxios.put(`config-services/api/posts/set-image`, data);
    },
    getById: (id) => {
        return httpAxios.get(`config-services/api/posts/get-by-id/${id}`);
    },
    getAll: () => {
        return httpAxios.get(`config-services/api/posts/get-all`);
    },
    update: (id, post) => { 
        return httpAxios.put(`config-services/api/posts/update/${id}`, post);
    },
    switchStatus: (id) => {
        return httpAxios.put(`config-services/api/posts/switch-status/${id}`);
    },
    trash: (id) => {
        return httpAxios.put(`config-services/api/posts/trash/${id}`);
    },
    delete: (id) => {
        return httpAxios.delete(`config-services/api/posts/delete/${id}`);
    },
};

export default Postservice;